### PR TITLE
Feature/gcm update

### PR DESCRIPTION
**Pull request by jrconlin**
* Added test for gcm send-retry
* merged fix for 202 on non-routable.
Requires PR #207 

@kitcambridge r?

### CODE REVIEW

File: src/github.com/mozilla-services/pushgo/simplepush/proprietary_ping.go [hunk 1]
```diff
@@ -169,11 +169,16 @@ func NewGCMPing() (r *GCMPing) {
 	return r
 }
 
+type GCMClient interface {
+	// for testing, based off minimial requirements from http.Client
+	Do(*http.Request) (*http.Response, error)
```
Comment (kitcambridge) [inline src/github.com/mozilla-services/pushgo/simplepush/proprietary_ping.go]: Nice refactoring!

File: src/github.com/mozilla-services/pushgo/simplepush/proprietary_ping_test.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+type rbody struct {
+	io.Reader
+}
+
+func (r rbody) Close() error {
+	return nil
+}
+
+// Generate a fake body ReadCloser
+func respBody(data string) io.ReadCloser {
```
Comment (kitcambridge) [inline src/github.com/mozilla-services/pushgo/simplepush/proprietary_ping_test.go]: Check out `ioutil.NopCloser`.

**Comment by kitcambridge** [comment]
Looking good, @jrconlin! I think we can schedule this for the v1.5 train, unless you'd prefer to wait—the tests help with coverage, and we won't have pingers enabled, anyway.

File: src/github.com/mozilla-services/pushgo/simplepush/proprietary_ping_test.go (hunk 2 above)
Comment (jrconlin) [inline src/github.com/mozilla-services/pushgo/simplepush/proprietary_ping_test.go]: neat! righto, replacing.

**Comment by jrconlin** [comment]
I'm fine either way. It can wait 'til 1.6 if we want to keep 1.5 focused, or we can add it as shoring up an existing feature.

**Comment by kitcambridge** [comment]
Let's land it in 1.5. Mind fixing up the change log and read me?

**Comment by kitcambridge** [comment]
r+

**Comment by kitcambridge** [comment]
r+

**Pull request merged**